lib/Site.js

```diff
@@ -120,26 +143,39 @@ Site.prototype.createPageData = function (config) {
     pageTemplate: config.pageTemplate,
     baseUrlMap: this.baseUrlMap,
     asset: {
-      bootstrap: path.relative(path.dirname(resultPath), path.join(this.siteAssetsDestPath, 'css', 'bootstrap.min.css')),
-      highlight: path.relative(path.dirname(resultPath), path.join(this.siteAssetsDestPath, 'css', 'github.min.css')),
-      markbind: path.relative(path.dirname(resultPath), path.join(this.siteAssetsDestPath, 'css', 'markbind.css')),
-      vue: path.relative(path.dirname(resultPath), path.join(this.siteAssetsDestPath, 'js', 'vue.min.js')),
-      vueStrap: path.relative(path.dirname(resultPath), path.join(this.siteAssetsDestPath, 'js', 'vue-strap.min.js')),
-      setup: path.relative(path.dirname(resultPath), path.join(this.siteAssetsDestPath, 'js', 'setup.js')),
-    }
+      bootstrap: path.relative(path.dirname(resultPath),
+                               path.join(this.siteAssetsDestPath, 'css', 'bootstrap.min.css')),
+      highlight: path.relative(path.dirname(resultPath),
+                               path.join(this.siteAssetsDestPath, 'css', 'github.min.css')),
+      markbind: path.relative(path.dirname(resultPath),
+                              path.join(this.siteAssetsDestPath, 'css', 'markbind.css')),
+      vue: path.relative(path.dirname(resultPath),
+                         path.join(this.siteAssetsDestPath, 'js', 'vue.min.js')),
+      vueStrap: path.relative(path.dirname(resultPath),
+                              path.join(this.siteAssetsDestPath, 'js', 'vue-strap.min.js')),
+      setup: path.relative(path.dirname(resultPath),
+                           path.join(this.siteAssetsDestPath, 'js', 'setup.js')),
+    },
   });
-}
+};
 
 Site.prototype.collectBaseUrl = function () {
   return new Promise((resolve, reject) => {
     // console.log('Root: ', this.rootPath)
-    let candidates = walkSync(this.rootPath, {directories: false}).filter(x => x.endsWith('site.json')).map(x => path.resolve(x));
-    let lookUp = candidates.reduce((pre, now) => {
-      pre[path.dirname(now)] = true
+    const candidates
+      = walkSync(this.rootPath, { directories: false })
+        .filter(x => x.endsWith('site.json'))
+        .map(x => path.resolve(x));
+
+    if (!candidates.length) return reject();
```

Comment:
@Gisonrg What do you think about this?

Reply:
I don't think we need to reject this if the candidates is empty. If we early return here then `this.baseUrlMap` would be an undefined value.
It does not give us much performance improvement. If the candidates is empty, the following `lookUp` will not run for the empty array.
If we concern about there could be no `site.json`, if the `site.json` in root is missing then `readSiteConfig()` will complain about it. So here we should have at least one `site.json` in the candidate.

Reply:
So `'reject' is defined but never used` should be ignored?

Reply:
If we just want to avoid the unused `reject`, collectBaseUrl is actually a sync function. No async operation involved. So instead of creating a new Promise, we could refactor it as:

```
Site.prototype.collectBaseUrl = function () {
  const candidates
    = walkSync(this.rootPath, { directories: false })
      .filter(x => x.endsWith('site.json'))
      .map(x => path.resolve(x));

  this.baseUrlMap = candidates.reduce((pre, now) => {
    // eslint-disable-next-line no-param-reassign
    pre[path.dirname(now)] = true;
    return pre;
  }, {});

  return Promise.resolve();
};
```
I think it should work as expected.